### PR TITLE
Refactor and change default scenarios

### LINKED ISSUE #13: Refactor old function arguments, that don't comply with the object_name_lintr = camelCase

### LINKED ISSUE #16: Improve documentation of functions

### LINKED ISSUE #56: No citable source for GDP data.
The documentation of [`calcGDP()`](https://github.com/pik-piam/mrdrivers/blob/eb1da620d641ff8df83dbe8861a0e320400f3527/R/calcGDP.R) does not contain any citable source for GDP data.  Like one would need for _doing science_.

- https://github.com/pik-piam/mrdrivers/blob/eb1da620d641ff8df83dbe8861a0e320400f3527/R/calcGDP.R#L49
And what happens if `GDPCalib` is `NULL`?

- https://github.com/pik-piam/mrdrivers/blob/eb1da620d641ff8df83dbe8861a0e320400f3527/R/calcGDP.R#L65
And what happens if `GDPPast` is `NULL`?

- https://github.com/pik-piam/mrdrivers/blob/eb1da620d641ff8df83dbe8861a0e320400f3527/R/calcGDP.R#L82
And what happens if `GDPFuture` is `NULL`?

- https://github.com/pik-piam/mrdrivers/blob/eb1da620d641ff8df83dbe8861a0e320400f3527/R/calcGDP.R#L85
Besides being completely insufficient as a source, the link is also dead.

I would really like to know how https://github.com/pik-piam/mrdrivers/commit/ec02c71c7599a5bd2025ef8866d68390214399be was supposed to provide anything citable.

-----

https://github.com/pik-piam/mrdrivers/blob/eb1da620d641ff8df83dbe8861a0e320400f3527/R/calcGDP.R#L25-L28
At least somebody had fun.

### LINKED ISSUE #57: Insufficient citable sources for Population data.
The documentation of [`calcPopulation()`](https://github.com/pik-piam/mrdrivers/blob/eb1da620d641ff8df83dbe8861a0e320400f3527/R/calcPopulation.R) does not contain usable citations for most sources of population data.  Like one would need for _doing science_.

- https://github.com/pik-piam/mrdrivers/blob/eb1da620d641ff8df83dbe8861a0e320400f3527/R/calcPopulation.R#L23
And what happens if `PopulationCalib` is `NULL`?

- https://github.com/pik-piam/mrdrivers/blob/eb1da620d641ff8df83dbe8861a0e320400f3527/R/calcPopulation.R#L39
And what happens if `PopulationPast` is `NULL`?

- https://github.com/pik-piam/mrdrivers/blob/eb1da620d641ff8df83dbe8861a0e320400f3527/R/calcPopulation.R#L49
  And what happens if `PopulationFuture` is `NULL`?
  - https://github.com/pik-piam/mrdrivers/blob/eb1da620d641ff8df83dbe8861a0e320400f3527/R/calcPopulation.R#L52-L53
    OK.
  - https://github.com/pik-piam/mrdrivers/blob/eb1da620d641ff8df83dbe8861a0e320400f3527/R/calcPopulation.R#L54-L57
    And where are those?
  - https://github.com/pik-piam/mrdrivers/blob/eb1da620d641ff8df83dbe8861a0e320400f3527/R/calcPopulation.R#L56
    Good thing [this got deleted](https://github.com/pik-piam/mrcommons/commit/4a21ac83f991e7ab26e7a9a321338e5cde74e041#diff-d6d7f28db97ea00150cadabc0abfb4b0b71e43bd4944eb8ad5cf38121b1b4f89). 

I would really like to know how https://github.com/pik-piam/mrdrivers/commit/ec02c71c7599a5bd2025ef8866d68390214399be was supposed to provide anything citable.

### LINKED ISSUE #91: 2005 → 2017 conversion missing for per-capita GDP figures in James and James2019

### PR DESCRIPTION
**Pull request by johanneskoch94**
- Replace SSP2EU with a placeholder, as it no longer a desired scenario, but required in the input preparation pipeline. Remove completely in next release.
- Add ADB sources and scenarios (not returned by default yet).
- Urban pop share scenarios are improved, so as to never exceed 1, even with bezier extensions.
- SSP labour scenarios are now also filled in with UN_popDiv data.
- SSP population scenarios now use short term growth rates from PEAP-UN_PopDiv, instead of PEAP-MI, improving the short-term projections of the smaller countries.
- Refactor function arguments that did not comply with lintr rules (fixes https://github.com/pik-piam/mrdrivers/issues/13).
- Mark "scenario.indicator" naming option as deprecated and remove in next release.
- Fix all linter warnings.
- Improve scenario documentation and references in vignette (fixes https://github.com/pik-piam/mrdrivers/issues/56 and fixes https://github.com/pik-piam/mrdrivers/issues/57).
- Improve and simplify documentation (fixes https://github.com/pik-piam/mrdrivers/issues/16).
- Remove "unit" from most functions, assuming instead that the default dollar unit is used throughout the package.
- Clean up use of James data set for filling and backwards extension to 1960. These two functions are now cleanly separated from each other. Also, use of the James2019 dataset is dropped (fixes https://github.com/pik-piam/mrdrivers/issues/62), due to the fact that there is no reference for this dataset. This does not affect the scenarios much since the dataset is mainly used for its growth rates, and they shouldn't differ to much.
- Refactor toolReduce and toolFillWith. toolFillWith now writes to the log which countries are filled in.
- Export some of the more useful tool functions.
- Speed up bezier extension (fixes https://github.com/pik-piam/mrdrivers/issues/78).
- Remove MI from future datasets and make it clear that we assume constant extrapolation for most of the MI countries.
- Fixes https://github.com/pik-piam/mrdrivers/issues/52.
- Fixes #91 
- Add warnings when converting to non-default base-year units (the current base year being 2017). These conversions (within calcGDP and caldGDPpc) use only the US deflator for all countries, and neglect any changes in PPPs or MERs. Standard conversion would not guarantee that relationships between countries (an important charateristic of the scenario design) remain same.
- Make choice of graphics device explicit when computing the bezier extensions (fixes https://github.com/pik-piam/mrdrivers/issues/79).

### CODE REVIEW

**Comment by johanneskoch94** [comment]
Hey @0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q. I would like to close issues #56 and #57 with this PR. Would you take a look and let me know if you agree? The references are [here](https://github.com/pik-piam/mrdrivers/pull/89/files#diff-bad96fefad64eb22e7883093451886880ac052bd47156c60abc856cb08ac2cfd), and this vignette is linked to in the descriptions of the calc functions, and also available on the website.

**Comment by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q** [comment]
Nice.  Personally, I would prefer a link to the Zenodo DOI instead of a direct download link where possible (i.e. https://doi.org/10.5281/zenodo.4421504 instead of https://zenodo.org/record/4421504/files/MissingIslands.zip).

**Comment by LaviniaBaumstark** [comment]
I hope that deleting SSP2EU will not cause any problems

**Comment by johanneskoch94** [comment]
> I hope that deleting SSP2EU will not cause any problems

A SSP2EU scenario is still returned, a placeholder of sorts, that is an exact copy of SSP2. I think the downstream deletion of SSP2EU from the pipeline will take a bit more work...

File: R/calcGDPFuture.R [hunk 1]
```diff
@@ -1,165 +1,122 @@
...
-    "SSPs"   = calcOutput("InternalGDPFutureSSPs", unit = unit, aggregate = FALSE, supplementary = TRUE),
-    "SSP2"   = calcOutput("InternalGDPFutureSSP2", unit = unit, aggregate = FALSE, supplementary = TRUE),
-    "SSP2EU" = calcOutput("InternalGDPFutureSSP2EU", unit = unit, aggregate = FALSE, supplementary = TRUE),
-    "SDPs"   = calcOutput("InternalGDPFutureSDPs", unit = unit, aggregate = FALSE, supplementary = TRUE),
-    "MI"     = calcOutput("InternalGDPMI", unit = unit, aggregate = FALSE, supplementary = TRUE),
-    stop("Bad input for calcGDPFuture. Invalid 'GDPFuture' argument.")
+    futureData,
+    "SSPs"   = readSource("SSP", "gdp"),
+    "SSP2"   = readSource("SSP", "gdp", "SSP2"),
+    "SSP2EU" = setNames(readSource("SSP", "gdp", "SSP2"), "SSP2EU"),
```
Comment (LaviniaBaumstark) [inline R/calcGDPFuture.R:19]: this still exists?

**Review by LaviniaBaumstark — APPROVED** [review]

File: R/calcGDPFuture.R (hunk 1 above)
Comment (johanneskoch94) [reply R/calcGDPFuture.R:19]: Yes, but only as a placeholder. Just so that the input data pipeline isn't broken.

**Pull request merged**